### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jyn514**
It was looking lonely :(

r? @hdamron17

### CODE REVIEW

**Comment by hdamron17** [comment]
Yeah this makes you sound a lot less like a hermit.

One suggestion for the instructions on writing runner tests is that you might make it more obvious in `run_one` what `// code: x` and the other parameterized annotations do (I was pretty sure had had to dig into `utils::assert_code` to confirm).
https://github.com/jyn514/rcc/blob/4b0ee92a0d68abd54564a24eaf59248f8adf47f4/tests/runner.rs#L31.

**Comment by jyn514** [comment]
Ok, I'll add some comments in runner.rs

File: tests/runner.rs [hunk 1]
```diff
@@ -38,19 +38,29 @@ fn run_one(path: &path::Path) -> Result<(), io::Error> {
...
+        // the test shouldn't compile
         "// fail" | "// compile-fail" | "// compile-error" => utils::assert_compile_error,
+        // it should compile, run, and exit successfully
         "// succeeds" => utils::assert_succeeds,
+        // it should compile successfully then crash at runtime
         "// crash" => utils::assert_crash,
+        // tests can only be ignored if they have an issue open on github
         "// ignore" => panic!("ignored tests should have an associated issue"),
         line => {
+            // `code: x` - it should compile, run, and exit with code 1
```
Comment (Byter09) [inline tests/runner.rs:54]: This should probably say `[..] and exit with code x`, right?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -9,24 +9,47 @@ the compiler will not break backwards compatibility with C.
...
+Writing unit tests for nested types can get tedious very quickly.
+Writing unit tests for codegen is not really possible,
+since we want to see how the program behaves at runtime,
+and running arbitrary C code in-process is a recipe for disaster.
+
+Instead, rcc uses 'Runner tests', which are C files in
+`tests/runner-tests` that are compiled and run by `tests/runner.rs`.
+You can control the expected output using a comment at the top of the file
+(usually something like `// succeeds` or `code: 1`).
+See [`runner.rs:run_one`](https://github.com/jyn514/rcc/blob/4b0ee92a0d68abd54564a24eaf59248f8adf47f4/tests/runner.rs#L31)
```
Comment (jyn514) [inline CONTRIBUTING.md:49]: I'll update this perma-link when we decide `runner.rs` looks good.

File: tests/runner.rs (hunk 1 above)
Comment (jyn514) [reply tests/runner.rs:54]: Yes it should!

Comment (hdamron17) [inline tests/runner.rs]: Should be `exit with code x`, no?

Comment (hdamron17) [reply tests/runner.rs]: Oh wait you already caught it

**Comment by hdamron17** [comment]
Looks good to me

**Pull request merged**